### ISSUE TITLE
Likely typo error in the algorithm as shown in the paper

### ISSUE THREAD

**Issue by tdeboissiere** [jayanthkoushik/sgd-feedback#2]
In eve.py, line 32:

ch_fact_lbound = K.switch(K.greater(loss, loss_prev), 1+self.thl, 1/(1+self.thu))
ch_fact_ubound = K.switch(K.greater(loss, loss_prev), 1+self.thu, 1/(1+self.thl))

This implies that if loss > loss_prev, then ch_fact_lbound = 1 + self.thk

If I'm not mistaken, this is the opposite of what's written in the paper.

I ran into this issue when I tried to implement the algorithm independently. If you replace K.greater by K.lesser (following what's written in the paper) then the loss quickly diverges.

For what it's worth, I have a somewhat simpler keras implementation of Eve here: https://github.com/tdeboissiere/DeepLearningImplementations/blob/master/Eve/Eve.py

**Comment by jayanthkoushik** [comment]
Yup! You're right - it's a typo in the paper. Thank you very much for pointing this out - we'll rectify this as soon as possible.

**Comment by tdeboissiere** [comment]
By the way, what were the best parameters (learning rate, decay) for Eve on CIFAR 100 ?

I am unable to reproduce your results on this dataset.

**Comment by jayanthkoushik** [comment]
Hi @tdeboissiere: for that experiment the best parameters for Eve are learning rate 0.0001 and decay 0.0001. What results did you get?

**Comment by tdeboissiere** [comment]
Eve actually did worse than Adam and SGD. My best guess is that without a lower learning rate + annealing, training is quite hard. I also had to switch on BatchNorm to get any optimizer to work.

I tried both you implementation and mine with much the same results.

What were the best parameters for SGD and Adam ?

**Comment by jayanthkoushik** [comment]
Did you try the learning rates I mentioned?

The best parameters for SGD, Adam:
SGD Nesterov: lr 0.01, decay 0.0001
Adam: lr 0.0001, decay - turns out to not have been saved because of a previous bug in Keras. But I would imagine it's the same as for Eve: 0.0001. I will try to run the cross-validation again when I can

**Comment by tdeboissiere** [comment]
I am going to run the experiments on CIFAR100 with the best parameters you found. Did you use BatchNorm in the end ?

**Comment by jayanthkoushik** [comment]
No, I didn't. The exact model is in src/models.py (get_big_cnn_model)

**Comment by tdeboissiere** [comment]
I reran the CIFAR100 experiment ONCE for each optimizer (cf. https://github.com/tdeboissiere/DeepLearningImplementations/tree/master/Eve) and this time, Eve outperforms the other 2.
